src/queue.ts

```diff
@@ -0,0 +1,1052 @@
+                  ? codec.read(message.Body)
+                  : (message.Body as unknown as Message),
+            })),
+          };
+        },
+      },
+      asl: (call, context) => {
+        const queueUrl = this.queueUrl;
+        const serializer = this.serializer;
+        const input = call.args[0]?.expr;
+
+        this.resource.grantConsumeMessages(context.role);
+        const messagesHeap = context.newHeapVariable();
+
+        if (input === undefined) {
+          return context.evalContext(call, ({ addState }) => {
+            addState(receiveAndParse(undefined));
+            return {
+              jsonPath: messagesHeap,
+            };
+          });
+        } else {
+          return context.evalExprToJsonPathOrLiteral(
+            input,
+            (input, { addState }) => {
+              if (
+                ASLGraph.isJsonPath(input) ||
+                input.value === null ||
+                typeof input.value !== "object"
+              ) {
+                // Because of limitations in `Parameters` - need support for `Parameters.$`
+                throw new SynthError(
+                  ErrorCodes.Invalid_Input,
+                  `the 'input' to receiveMessage must be an object literal`
+                );
+              }
+              addState(receiveAndParse(input.value));
+              return {
+                jsonPath: messagesHeap,
+              };
+            }
+          );
+        }
+
+        function receiveAndParse(
+          request: object | undefined
+        ): ASLGraph.SubState {
+          if (
+            serializer.dataType !== DataType.Json &&
+            serializer.dataType !== DataType.String
+          ) {
+            throw new SynthError(
+              ErrorCodes.Unexpected_Error,
+              `unknown DataType '${serializer.dataType}' is not supported by SQS by Step Functions`
+            );
+          }
+
+          return {
+            startState: "receive",
+            states: {
+              receive: {
+                Type: "Task",
+                Resource: "arn:aws:states:::aws-sdk:sqs:receiveMessage",
+                ResultPath: messagesHeap,
+                Parameters: {
+                  ...(request ? request : {}),
+                  QueueUrl: queueUrl,
+                },
+                Next:
+                  serializer.dataType === DataType.Json
+                    ? "parseIfMessages"
+                    : ASLGraph.DeferNext,
+              },
+              ...(serializer.dataType === DataType.Json
+                ? {
+                    parseIfMessages: {
+                      Type: "Choice",
+                      Choices: [
+                        {
+                          IsPresent: true,
+                          Variable: `${messagesHeap}.Messages`,
+                          Next: "parse",
+                        },
+                      ],
+                      Default: ASLGraph.DeferNext,
+                    },
```

Comment:
we remove all unreachable states, so you can add the state to the graph and it will be removed later. This is fine too.